source/globalCommands.py

```diff
@@ -4891,6 +4891,30 @@ def script_toggleApplicationsVolumeAdjuster(self, gesture: "inputCore.InputGestu
 	def script_toggleApplicationsMute(self, gesture: "inputCore.InputGesture") -> None:
 		appsVolume._toggleAppsVolumeMute()
 
+	@script(
+		description=_(
+			# Translators: Input help mode message for report language for caret command.
+			"Reports the language for the character where the caret is positioned. "
+			"Pressed twice, shows the character and its language in browse mode",
+		),
+		category=SCRCAT_SYSTEMCARET,
+		speakOnDemand=True,
+	)
+	def script_reeportCaretLanguage(self, gesture):
+		info = self._getTIAtCaret()
+		info.expand(textInfos.UNIT_CHARACTER)
+		curLanguage = self._getCurrentLanguageForTextInfo(info)
+		text = info.text
+		repeats = scriptHandler.getLastScriptRepeatCount()
+		if repeats == 0:
+			ui.message(languageHandler.getLanguageDescription(curLanguage))
```

Comment:
Please handle the case where the language is not recognized. I.e. report something like `f"Unknown language ({languageCode})"` or just `f"{languageCode}"`.